public/app/ai-intelligence.js

```
@@ -0,0 +1,430 @@
+    let total = 0;
+    let maxTotal = 0;
+    for (const [value, min, max, weight] of features) {
+      total += this._scoreFeature(value, min, max, weight);
+      maxTotal += weight;
+    }
+    return maxTotal > 0 ? total / maxTotal : 0;
+  }
+};
+
+// Export for both browser and Node.js (tests)
+if (typeof module !== 'undefined' && module.exports) {
```

ESLint

[error] 'module' is not defined.

(no-undef)